src/components/v5/common/ActionSidebar/partials/forms/ArbitraryTxsForm/partials/AddTransactionModal/ContractAddressInput.tsx

```diff
@@ -0,0 +1,48 @@
+import { isAddress } from 'ethers/lib/utils';
+import React, { useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { fetchArbiscanContract } from '~utils/arbitraryTransaction/fetchArbiscanContract.ts';
+import { formatText } from '~utils/intl.ts';
+import FormInput from '~v5/common/Fields/InputBase/FormInput.tsx';
+
+import { validateContractAddress } from './consts.ts';
+import { MSG } from './translation.ts';
+
+export const ContractAddressInput = () => {
+  const { watch, setError, setValue, clearErrors, trigger } = useFormContext();
+
+  const contractAddressField = watch('contractAddress');
+
+  const [serverError, setServerError] = useState('');
+
+  useEffect(() => {
+    if (isAddress(contractAddressField)) {
+      getABIFromContractAddress(contractAddressField);
+    }
+
+    async function getABIFromContractAddress(contractAddress: string) {
+      const response = await fetchArbiscanContract(contractAddress);
+
+      if (response.status === '0') {
+        setServerError(response.result);
```

Comment:
This didn't catch if I didn't have the API key set.

Reply:
Hey, seems that this one is expected from API. It allow you to send couple of requests without API key:

<img width="489" alt="image" src="https://github.com/user-attachments/assets/587f22d8-6a00-4537-a3a8-0390b1d06e6e">

Reply:
Oh, in that case it wasn't showing the error for invalid address in that scenario (I was using the colony address)

Reply:
Yes, you’re right. Even though the API allows us to make calls without an API key (with certain limits), there was definitely a bug with the server validation. I’ve fixed it, and I hope the issue you mentioned will now be resolved with these changes. 🙌